### PR TITLE
Fix skip_optional_dependencies for post-1.60 cargo

### LINKED ISSUE #27: Bug: Tries too many / invalid combinations of features on beta/nightly Rust
I just added this to my CI. But I noticed when it runs on the Rust beta and nightly channels it tries a very odd combination of features. My crate has two optional dependencies, those are the only two features. The crate has `env_logger` and `structopt` as optional features.

On stable (only the relevant output):
```
$ cargo +stable build-all-features 2> /dev/null
    Building crate=udp-over-tcp features=[]
    Building crate=udp-over-tcp features=[env_logger]
    Building crate=udp-over-tcp features=[structopt]
    Building crate=udp-over-tcp features=[env_logger, structopt]
```

The above is exactly what I expect. But it does the following on beta:
```
 $ cargo +beta build-all-features 2> /dev/null
    Building crate=udp-over-tcp features=[]
    Building crate=udp-over-tcp features=[env_logger]
    Building crate=udp-over-tcp features=[structopt]
    Building crate=udp-over-tcp features=[env_logger]
    Building crate=udp-over-tcp features=[structopt]
    Building crate=udp-over-tcp features=[env_logger, structopt]
    Building crate=udp-over-tcp features=[env_logger, env_logger]
    Building crate=udp-over-tcp features=[env_logger, structopt]
    Building crate=udp-over-tcp features=[structopt, env_logger]
    Building crate=udp-over-tcp features=[structopt, structopt]
    Building crate=udp-over-tcp features=[env_logger, structopt]
    Building crate=udp-over-tcp features=[env_logger, structopt, env_logger]
    Building crate=udp-over-tcp features=[env_logger, structopt, structopt]
    Building crate=udp-over-tcp features=[env_logger, env_logger, structopt]
    Building crate=udp-over-tcp features=[structopt, env_logger, structopt]
    Building crate=udp-over-tcp features=[env_logger, structopt, env_logger, structopt]
```

This is not what I would expect it to try with only two features :sweat_smile:

### LINKED ISSUE #28: Support chunking
This tool tends to fill up target directories really quickly. For large projects like ICU4X, this puts us in danger of hitting CI size limits. Furthermore, it just takes pretty long and it would be nice to be able to make use of parallelism.

A typical technique in CI systems is to support "chunking", where you have something like `--chunks 3 --chunk 1`, which splits the list of jobs into three chunks, and directs it to run just the jobs for the first one, so you can have the chunks get run on separate runners.

Thoughts?

### LINKED ISSUE #29: skip_optional_dependencies is broken with 1.60
With 1.60 and [`dep:` syntax](https://doc.rust-lang.org/nightly/cargo/reference/features.html#optional-dependencies), `cargo metadata` now always reports a feature of the form `"foo": ["dep:foo"]` for optional dependencies. This leads to all optional dep combos being tested even with `skip_optional_dependencies, which bloats runtime significantly.

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #27, fixes #29

@frewsxcv can we get a release with this soon?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -32,6 +32,8 @@ fn test_all_features_for_package(
 ) -> Result<TestOutcome, Box<dyn error::Error>> {
     let feature_sets = crate::features_finder::fetch_feature_sets(package);
 
+    println!("{:?}", feature_sets);
```
Comment (sffc) [inline src/lib.rs:35]: Nit: Debugging, right?

Comment (frewsxcv) [inline src/lib.rs:35]:
```suggestion

```

**Review by frewsxcv — APPROVED** [review]

**Comment by frewsxcv** [comment]
@Manishearth I just published 1.7.0 with this

**Comment by Manishearth** [comment]
Sweet thanks!

**Pull request merged**